actionpack/lib/action_controller.rb

```diff
@@ -54,6 +54,12 @@ module ActionController
   autoload :TemplateAssertions, "action_controller/test_case"
 end
 
+ActiveSupport.on_load(:after_initialize) do
+  if config.eager_load
+    ActionController::Base.descendants.each(&:action_methods)
```

Comment:
How about subclasses on `ActionController::Api`, `ActionController::Metal` and `ActionMailer::Base`? I think we should use `ActionController::Metal` here and do the same with `ActionMailer::Base` in its railtie.